src/PolyhedralGeometry/Polyhedron/standard_constructions.jl

```diff
@@ -411,8 +411,32 @@ julia> length(vertices(product(T,S)))
 ```
 """
 function product(P::Polyhedron{T}, Q::Polyhedron{U}) where {T<:scalar_types,U<:scalar_types}
-  V, f = _promote_scalar_field(coefficient_field(P), coefficient_field(Q))
-  return Polyhedron{V}(Polymake.polytope.product(pm_object(P), pm_object(Q)), f)
+  if lineality_dim(P) == 0 && lineality_dim(Q) == 0
+    V, f = _promote_scalar_field(coefficient_field(P), coefficient_field(Q))
+    return Polyhedron{V}(Polymake.polytope.product(pm_object(P), pm_object(Q)), f)
+  end
+
+  # Take the product without the linealities
+  Pmod, LP = peel_lineality(P)
+  Qmod, LQ = peel_lineality(Q)
+  PQmod = product(Pmod, Qmod)
+  K = coefficient_field(PQmod)
+
+  # Construct the product of the linealities
+  n = ambient_dim(P)
+  m = ambient_dim(Q)
+  LPQ = vcat(Vector{elem_type(K)}[vcat(lp, zeros(elem_type(K), m)) for lp in LP],
+    Vector{elem_type(K)}[vcat(zeros(elem_type(K), n), lq) for lq in LQ])
```

Comment:
```suggestion
  LPQ = block_diagonal_matrix([matrix(K, LP), matrix(K, LQ)])
```
I think this should work.